Added a limitation regarding reconnection (#1306)
mulesoft/docs-connectors: sap/5.3/modules/ROOT/pages/sap-connector-reference.adoc

== Limitations

* SNC is not supported for use in Mule apps that deploy SAP Connector in CloudHub and Anypoint Runtime Fabric.

> * For versions prior to SAP Connector 5.3.0, multiple applications deployed on the same Mule runtime engine with the same PROGRAM_ID, GATEWAY_SERVICE, and GATEWAY_HOST cannot be created by the JCo library. If you attempt to create two servers with the same PROGRAM_ID, GATEWAY_SERVICE, and GATEWAY_HOST, only one server is created. Some workarounds are to use two different Mule runtime engines or use the same PROGRAM_ID with two different GATEWAY_HOSTs.

Reviewer: ```suggestion
* For SAP Connector versions earlier than 5.3.0, multiple applications deployed on the same Mule runtime engine with the same PROGRAM_ID, GATEWAY_SERVICE, and GATEWAY_HOST cannot be created by the JCo library. If you create two servers with the same PROGRAM_ID, GATEWAY_SERVICE, and GATEWAY_HOST, only one server is created. To work around this, you can use two different Mule runtime engines or use the same PROGRAM_ID with two different GATEWAY_HOSTs.
```

Reviewer: So, there is a mix of application and server.  Is this really:
For versions prior to SAP Connector 5.3.0, multiple servers with the same PROGRAM_ID, GATEWAY_SERVICE, and GATEWAY_HOST cannot be created by the JCo library. 
?

* SAP Connector does not support Bitronix Transaction Manager.

> * The reconnection strategy on SAP Listeners (Function Listener and Document Listener) relies exclusively on the JCo library which means that the MULE mechanism has no impact on reconnections.

Reviewer: ```suggestion
* The reconnection strategy on SAP Listeners (Function Listener and Document Listener) relies exclusively on the JCo library, which means that the Mule reconnection mechanism has no impact on reconnections.
```

> * The JCO Server that runs under the hood, manages the reconnection when there is a connection issue. The JCo RFC provider recognizes a communication error between the Java server and the ABAP gateway and attempts to reconnect. If the first reconnection fails, the next reconnection attempt occurs after waiting for one second. If the reconnection fails again, the reconnection wait time doubles from one second to two seconds, until eventually it reaches the default maximum waiting period of 3600 seconds, or one hour. The maximum reconnection timeout can be set using the `-Djco.server.max_startup_delay=<reconnect delay in seconds>` JMV parameter.

Reviewer: ```suggestion
* The JCO Server that runs in the background manages the reconnection when there is a connection issue. The JCo RFC provider recognizes a communication error between the Java server and the ABAP gateway and attempts to reconnect. If the first reconnection fails, the next reconnection attempt occurs after waiting for one second. If the reconnection fails again, the reconnection wait time doubles from one second to two seconds, until eventually it reaches the default maximum waiting period of 3600 seconds, or one hour. Set the maximum reconnection timeout using the `-Djco.server.max_startup_delay=<reconnect delay in seconds>` JMV parameter.
```

Workarounds:
. Use two different Mule runtime engines.
. Use the same PROGRAM_ID but with two different GATEWAY_HOSTs.
. Starting with SAP Connector 5.3.0 this feature is supported and the user can add multiple flows with multiple sources using the same PROGRAM_ID, GATEWAY_SERVICE and GATEWAY_HOST.

* SAP Connector does not support Bitronix Transaction Manager. 

* The reconnection strategy on SAP Listeners(Function Listener, Document Listener) relies exclusively on the JCO library which implies that the MULE mechanism for reconnections has no impact.
* The JCO Server that runs  under the hood, in case of a connection issue it will manage the reconnection in the following ways:

 ** If a connection issue occurs, the JCo server retries immediately.

 ** If the connection is not restored immediately, the JCo server waits for one second and retries.

 ** If the connection is not restored after the second retry, the JCo server waits for two seconds and retries.

>  ** If the connection is still not restored, the JCo server retries at four seconds, and then again at eight seconds.

Reviewer 2: after that there isn't retries? If no please explain it here too

Author: I have modified those "ways"  based on this information. https://wiki.scn.sap.com/wiki/display/ASJAVA/SAP+JCo+FAQs#:~:text=It%20tries%20to%20reconnect%20in,be%20set%20using%20the%20%2DDjco. Check Q9.

* SAP Connector does not support Bitronix Transaction Manager. 

* The reconnection strategy on SAP (applies to operations only) is improved. With this enhancement, when a connection fails on the FunctionSource or DocumentSource, the JCo server automatically tries to reestablish a connection in the following way:

 ** If a connection issue occurs, the JCo server retries immediately.

>  ** If the connection is not restored immediately, the JCo server waits for one second and retries.

Reviewer 3: please mention that this relies exlusively on the jco mechanism, the mule mechanism has no effect here.

Reviewer 2: exactly.

" The reconnection strategy on SAP (applies to operations only) is improved" I'm not sure that that means

Author: Fixed.
